frameworks/helloworld/tests/test_discovery.py

```diff
@@ -31,5 +34,5 @@ def test_task_dns_prefix_points_to_all_tasks():
     # Assert that DiscoveryInfo is correctly set on tasks.
     assert(all(p["info"]["discovery"]["name"] == "hello-0" for p in pod_info))
     # Assert that the hello-0.hello-world.mesos DNS entry points to the right IP.
-    spin.time_wait_noisy(lambda: (
-        plan.get_deployment_plan(PACKAGE_NAME).json()['status'] == 'COMPLETE'))
+    shakedown.wait_for(lambda: (
+        plan.get_sidecar_plan(PACKAGE_NAME).json()['status'] == 'COMPLETE'))
```

Comment:
I think your problem is here. You are waiting for a plan named `sidecar` but `discovery.yml` only has `deploy` specified and `recovery` implicitly.